---
title: 'Class 14: Transcriptomics and the analysis of RNA-Seq data'
author: "Gayatri Mainkar"
date: "2/20/2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Import today's data

The data for this hands-on session comes from a published RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014).

Glucocorticoids are used, for example, by people with asthma to reduce inflammation of the airways. The anti-inflammatory effects on airway smooth muscle (ASM) cells has been known for some time but the underlying molecular mechanisms are unclear.

```{r}
counts <- read.csv("data/airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("data/airway_metadata.csv", stringsAsFactors = FALSE)
```

How many genes are we dealing with?

```{r}
nrow(counts)
```

We want to calculate the average counts per gene for the control samples and for the drug treated samples. We will then compare these values to see if there is a difference overall between control and treated samples for a given gene.

```{r}
control_id <- metadata$id[metadata$dex == "control"]
control.mean <- rowSums( counts[ ,control_id] )/length(control_id)
names(control.mean) <- counts$ensgene

treated_id <- metadata$id[metadata$dex == "treated"]
treated.mean <- rowSums( counts[ ,treated_id] )/length(treated_id)
names(treated.mean) <- counts$ensgene
```

```{r}
meancounts <- data.frame(control.mean, treated.mean)
plot.default(meancounts, log = "xy")
```

We can find candidate differentially expressed genes by looking for genes with a large change between control and dex-treated samples. We usually look at the log2 of the fold change, because this has better mathematical properties.

Here we calculate log2foldchange, add it to our meancounts data.frame

```{r}
meancounts$log2fc <-log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

Remove any gene that has zero avergae counts.

```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)
```

How many genes do I have left?
```{r}
nrow(mycounts)
```

```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)

sum(up.ind)
sum(down.ind)
```

## Do this properly with DESeq2

```{r}
library("DESeq2")
```

Setup input for DESeq by specifying countdata, metadata and the experiment design (i.e. which comparisons we are interested in)

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, colData=metadata, design=~dex, tidy=TRUE)
dds
```
```{r}
dds <- DESeq(dds)
```
```{r}
res <- results(dds)
```

## Plotting a Volcano plot
```{r}
# Setup our custom point color vector 
mycols <- rep("gray", nrow(res))
mycols[ abs(res$log2FoldChange) > 2 ]  <- "red" 

inds <- (res$padj < 0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

#Volcano plot with custom colors 
plot( res$log2FoldChange,  -log(res$padj), 
 col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

abline(v=c(-2,2), col="gray", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```

```{r}
biocLite("EnhancedVolcano")
```

